hypothesis-python/tests/cover/test_lookup_py37.py

```diff
@@ -0,0 +1,212 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2020 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import annotations
+
+import collections
+import collections.abc
+import contextlib
+import sys
+
+import pytest
+
+from hypothesis import assume, given, infer
+
+# This test file is skippped on Python <= 3.6, where `list[Elem]` is always a
+# SyntaxError.  On Python 3.7 and 3.8, `from __future__ import annotations` means
+# that the syntax is supported; but the feature fails at runtime.  On Python
+# 3.9 and later, it should all work.
+#
+# For details, see https://www.python.org/dev/peps/pep-0585/
+
+if sys.version_info < (3, 9):
```

Comment:
Not sure if this line is consistent with the filename and the docs.

Reply:
- On Python 3.6, `conftest.py` tells pytest not to collect this file (the annotations would be evaluated, and raise errors)
- On 3.7 and 3.8, the `pytestmark` means that all tests are expected to fail - `from __future__ import annotations` makes the new style work syntactically but not semantically
- On 3.9+, the tests should pass

Reply:
ok, got it 👍